src/msg/async/rdma/Infiniband.cc

```diff
@@ -964,6 +964,10 @@ Infiniband::~Infiniband()
     ibv_destroy_srq(srq);
   delete memory_manager;
   delete pd;
+  if (device_list) {
+    delete device_list;
+    device_list = nullptr;
```

Comment:
no need to reset member variable in dtor.

Reply:
@tchaikov I'm a kernel debugger before. So, I always keep the variable in some fixed value even it's not used anymore which could be good to do debug once hitting some unknown problem. If this is not preferred in ceph, I won't do this kind of change when changing others' code.

Reply:
i don't think it's necessary.

Reply:
Get it. I'll refine them.